src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2452,8 +2454,72 @@ public final void addBatch() throws SQLServerException {
         }
         checkClosed();
         discardLastExecutionResults();
-
+        
         int updateCounts[];
+        
+        localUserSQL = userSQL;
+        
+        try {
+            if (isInsert(localUserSQL) && connection.isAzureDW()) {
+                if (batchParamValues == null) {
+                    updateCounts = new int[0];
+                    loggerExternal.exiting(getClassNameLogging(), "executeBatch", updateCounts);
+                    return updateCounts; 
+                }
+
+                String tableName = parseUserSQLForTableNameDW(false, false);
+                ArrayList<String> columnList = parseUserSQLForColumnListDW();
+                ArrayList<String> valueList = parseUserSQLForValueListDW(false);
+                
+                String destinationTableName = tableName;
+                // Get destination metadata
+                try (SQLServerResultSet rs = ((SQLServerStatement) connection.createStatement())
```

Comment:
try with resources, but we don't handle any exceptions.

Reply:
see my comment above